controllers/member.controller.js

```diff
@@ -1,86 +1,95 @@
-import mongoose from "mongoose"
 
-//To add member to a channel and push the new member's id in the members array of Channel schema and save the channel.
-const addMemberToChannel = async (req, res, next) => {
+//To add member to a workspace and push the new member's id in the members array of Workspace schema and save the workspace.
+const addMemberToWorkspace = async (req, res, next) => {
     try {
-        const { channelId } = req.params;
+        const { workspaceId } = req.params;
         const { userId } = req.body;
         const user = await User.findById(userId);
         if (!user) {
             return res.status(404).json({ message: "User not found" });
         }
-        const channel = await Channel.findById(channelId);
-        if (!channel) {
-            return res.status(404).json({ message: "Channel not found" });
+        const workspace = await Workspace.findById(workspaceId);
+        if (!workspace) {
+            return res.status(404).json({ message: "Workspace not found" });
         }
         const newMember = new Members({
             userId: userId,
-            channelId: channelId
+            channelId: [],
+            workspaceId: workspaceId
         });
         await newMember.save();
-        channel.members.push(newMember._id);
-        await channel.save();
-        return res.status(201).json({ message: "User has joined the channel", member: newMember });
+        workspace.members.push(newMember._id);
+        await workspace.save();
+        return res.status(201).json({ message: "User has joined the workspace", member: newMember });
     } catch (error) {
         logger.error(error.message);
         return res.status(500).json({ message: "Error adding member", error });
     }
 };
 
-//To find and return the complete channel and user object about all the channels a user is part of.
-const getChannelMembers = async (req, res, next) => {
+//To find and return the complete workspace and user object about all the workspaces a user is part of.
+const getWorkspaceMembers = async (req, res, next) => {
     const { userId } = req.params;
     try {
-        const members = await Members.find({ userId }).populate('channelId');
+        const members = await Members.find({ userId }).populate('workspaceId');
 
         if (!members || members.length === 0) {
-            return res.status(404).json({ message: 'This user is not part of any Channel' });
+            return res.status(404).json({ message: 'This user is not part of any Workspace' });
         }
         const response = await Promise.all(members.map(async (member) => {
-            const channel = await Channel.findById(member.channelId).populate('owner');
+            const workspace = await Workspace.findById(member.workspaceId).populate('owner');
             const user = await User.findById(member.userId);
             return {
-                channel: channel,
+                workspace: workspace,
                 user: user,
```

Comment:
why sending user is required here , just asking

Reply:
Sir ,actually this is the member api we have made changes in after renaming channel to workspace.You had asked to send the channel and user object in the member get apis(in google doc) so earlier I had send the user object here to tell what other workspaces the user is part of.